pkg/proxy/apis/config/validation/validation.go

```diff
@@ -71,10 +72,14 @@ func Validate(config *kubeproxyconfig.KubeProxyConfiguration) field.ErrorList {
 	if config.ClusterCIDR != "" {
 		if config.FeatureGates[string(kubefeatures.IPv6DualStack)] {
 			cidrs := strings.Split(config.ClusterCIDR, ",")
-			for _, cidr := range cidrs {
-				if _, _, err := net.ParseCIDR(cidr); err != nil {
-					allErrs = append(allErrs, field.Invalid(newPath.Child("ClusterCIDR"), cidr, "must be a valid CIDR block (e.g. 10.100.0.0/16 or FD02::0:0:0/96)"))
+			if len(cidrs) == 2 {
```

Comment:
with IP6DualStack being "true" by default when it goes beta, this means that if the user has a single IPv4 CIDR in their config, it would fail this check and add an append an error on line 82. this makes the change not backwards compatible.

has this been discussed?

Reply:
I think that discussion is happening right now, see related PR
https://github.com/kubernetes/kubernetes/pull/86895
I've added a comment to the dual stack KEP to define the `skew strategy` too
https://github.com/kubernetes/enhancements/pull/1429/files#r368227247

Reply:
I agree with @neolit123 that a single-stack configuration should remain valid even once the dualstack feature graduates.

Feature gates are temporary and should be expected to eventually become enabled by default. The dualstack feature gate being enabled does not mean the cluster MUST be dualstack, just that it MAY be dualstack.

I would expect something more like this:

```
cidrs := strings.Split(config.ClusterCIDR, ",")
for _, cidr := range cidrs {
  ... validate cidr
}
switch {
  case len(cidrs) > 2:
    ... error, can't have more than two
  case len(cidrs) == 2 && feature disabled:
    ... error, can't have more than one with dual-stack feature disabled
  case len(cidrs) == 2 && feature enabled:
    ... validate cidrs are different families
}
```

Reply:
also, I'm not that familiar with the kube-proxy code, but `config.FeatureGates[string(kubefeatures.IPv6DualStack)]` doesn't seem like the correct way to check if the feature is enabled, since it depends on the default being `false`, which will not be accurate as the feature graduates to beta and is enabled by default unless explicitly overridden in config

Reply:
you are right, just one mistake I had is that we are not imposing any limitation on the number of CIDRs, there is no upper limit
https://github.com/kubernetes/enhancements/blame/master/keps/sig-network/20180612-ipv4-ipv6-dual-stack.md#L402

Reply:
https://github.com/kubernetes/kubernetes/pull/87353#discussion_r368568489

this seems a bug,  it also will not validate the config against the `flags`

https://github.com/kubernetes/kubernetes/blob/2808d7154dc820c2718d92eb2f000ffc5840b5d1/cmd/kube-proxy/app/server.go#L486-L491

will fix it in this PR too

Reply:
PTAL